orchestration/dagster_orchestration/hca_orchestration/support/typing.py

```diff
@@ -12,3 +14,13 @@ class HcaScratchDatasetName(str):
 
 
 make_python_type_usable_as_dagster_type(HcaScratchDatasetName, DagsterString)
+
+
+class MetadataType(str):
+    pass
+
+
+class MetadataTypeFanoutResult(NamedTuple):
+    scratch_dataset_name: HcaScratchDatasetName
+    metadata_type: MetadataType
+    is_file_metadata: bool = False
```

Comment:
🔧 this might be where we express the path in an earlier comment; rather than leaving it up to a function way downstream to care about whether it's file metadata or non-file metadata, we could determine that at the top and simplify the downstream logic.